src/targets/github.ts

```diff
@@ -361,97 +360,30 @@ export class GithubTarget extends BaseTarget {
     const uploadSpinner = ora(
       `Uploading asset "${name}" to ${this.githubConfig.owner}/${this.githubConfig.repo}:${release.tag_name}`
     ).start();
+
     try {
-      const file = readFileSync(path);
+      const file = createReadStream(path);
```

Comment:
Since most of the files we upload are small files, using `readFileSync` is actually safer and more efficient, especially compared to a read stream. This is a lesson we learned from `npm` folks when working on Yarn: https://github.com/yarnpkg/yarn/pull/3539

Reply:
Thanks for the reference, @BYK.

The change from `createReadStream` to `readFileSync` was not documented in #290, seemed like the motivation was just to get types to match (`string`) 🤔  which led to the broken asset uploads.

Later in https://github.com/getsentry/craft/pull/328#discussion_r752531205 we were discussing streaming... (at that time I had not realized that we changed from stream to read-all also in #290, relatively recently), and the concern was memory usage.

I haven't benchmarked, but I believe for our use case with Craft the performance difference probably doesn't matter?

> `readFileSync` is actually safer and more efficient

I can intuitively understand the "more efficient", but, I'm curious, why do you say it is also "safer"?

Reply:
> The reason this is OK is because pretty much any files this would
handle would fit neatly into memory (any npm packages MUST fit
into memory by definition, because of the way npm@<5 does extracts).
>
> If you really want to make doubleplus sure to minimize memory usage,
you could do an fs.stat to find the file size and then do heuristics
to only use streams for files bigger than MB.

Quoted from https://github.com/yarnpkg/yarn/pull/3539

Craft asset sizes, theoretically, don't really need to fit in memory (though we're probably not uploading any large assets in Sentry, and I don't even know what GitHub's limit is tbh).

The second paragraph applies -- we already do a `stat` on the asset file, we could choose between `readFileSync` and `createReadStream`, but not sure the added complexity is worth it. More code, more surface area for things to go wrong.

Reply:
>The change from createReadStream to readFileSync was not documented in #290, seemed like the motivation was just to get types to match (string) 🤔 which led to the broken asset uploads.

Yup, my bad for not making the reason for change explicit, sorry 😞 

Re in #328, I think we all decided that the asset sizes not fitting into memory was a distant potential issue, hence me being content with `readFileSync`.

>I haven't benchmarked, but I believe for our use case with Craft the performance difference probably doesn't matter?

You may be surprised 🙂 

>I can intuitively understand the "more efficient", but, I'm curious, why do you say it is also "safer"?

Because dealing with streams in Node is close to insanity. In a world dominated by promises nowadays, streams still rely on events and it is very easy to not handle a case where the stream errored out etc. Now you'd expect the Octokit library would handle that but we've already seen that this API was not stable enough with the bad typing information (yes this sounds like FUD but essentially, based on my past experience with streams, I'd try to avoid them at all costs).

Anyway, I just wanted to provide the context around it. This code worked with `createReadStream` for years so I don't think there's a strong case for going either way.

Reply:
Thanks @BYK, I really appreciate you taking the time to provide us so much insight 🙇 ❤️

Reply:
I cut https://github.com/getsentry/craft/issues/334 to whomever comes next maintaining Craft, no need to rush to make changes now.

Reply:
TIL streams <> node.